packages/react/src/components/cookieConsent/modal/Modal.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+
+import classNames from '../../../utils/classNames';
+import styles from '../CookieConsent.module.scss';
+import { CookieConsentContext, useCookieConsentUiTexts } from '../CookieConsentContext';
+import Content from '../content/Content';
+
+export function Modal(): React.ReactElement | null {
+  const cookieConsentContext = useContext(CookieConsentContext);
+  const shouldShowModal = !cookieConsentContext.hasUserHandledAllConsents();
+  const [isModalInitiallyShown] = useState<boolean>(shouldShowModal);
+  const [popupTimerComplete, setPopupTimerComplete] = useState<boolean>(false);
+  const popupDelayInMs = 500;
+  // if hasUserHandledAllConsents() was false at first and then later true, user must have saved consents.
+  const showScreenReaderSaveNotification = isModalInitiallyShown && !shouldShowModal;
```

Comment:
Would another alternative be to show the screenreader save notification when user clicks the save buttons, like done in the page version of the cookie consent?

Reply:
This component has no access to the buttons. The listener should be passed from here to child and from the to the buttons. That seems excessive prop drilling.